### PR TITLE
Bug 1633041 - Remove step parser v2

### PR DESCRIPTION
**Pull request by sarah-clements**
The only thing that's changed from the original pr #6546 (reverted last week) is the addition of logging and exception handling for redis, which I believe was causing the failure summary tab to show logs as still pending. I've also changed a few areas where I missed removing the `step__job` relationship (see the last commit). Not entirely sure which might have caused the increased redis memory consumption but it appears to have stabilized. 

FYI @camd

### CODE REVIEW

File: treeherder/webapp/api/jobs.py [hunk 1]
```diff
@@ -375,6 +375,7 @@ def list(self, request, project):
 
         return Response(response_body)
 
+    # TODO remove
```
Comment (armenzg) [inline treeherder/webapp/api/jobs.py:378]: Do you want to keep this comment?

**Comment by armenzg** [comment]
Did you get this running on Prototype and checked New Relic did show anything getting out of hand?
Was it through New Relic that we could notice the issue last week? (I vaguely remember)

File: treeherder/webapp/api/jobs.py (hunk 1 above)
Comment (sarah-clements) [reply treeherder/webapp/api/jobs.py:378]: yes. I'll remove it in the final pr to remove the step table.

**Comment by sarah-clements** [comment]
> Did you get this running on Prototype and checked New Relic did show anything getting out of hand?
> Was it through New Relic that we could notice the issue last week? (I vaguely remember)

I had it running on prototype from Friday through this morning and I didn't see any new errors. Yes, you saw the uncaught exception for redis memory on Thursday in New Relic. I can't be certain why the memory consumption had increased but it's been back to normal last I looked (both yesterday and today). The try/except block I've added fixed the issue where logs looked they were still parsing even though they had parsed (the uncaught redis cache exception caused the logs to never show as parsed even though they had been). I can leave on prototype for another day or so if you prefer?

**Comment by armenzg** [comment]
All good. If you're comfortable I'm comfortable.

**Comment by armenzg** [comment]
Reverted this.

**Pull request merged**